File: ria/java/gof/src/main/java/com/gof/pattern/prototype/before/GithubIssue.java
package com.gof.pattern.prototype.before;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class GithubIssue {

    private int id;

    private String title;

    private GithubRepository repository;

    public String getUrl() {
        return String.format("https://github.com/%s/%s/issues/%d",
                repository.getUser(),
                repository.getName(),
                this.getId());
    }
}

File: gof/src/main/kotlin/com/gof/pattern/prototype/before/GithubIssue.kt
package com.gof.pattern.prototype.before

class GithubIssue {
    var id: Int = 0
    var title: String = ""
    private var repository: GithubRepository = GithubRepository()

    fun getUrl(): String {
        return String.format(
            "https://github.com/%s/%s/issues/%d",
            repository.user,
            repository.name,
            this.id
        )
    }
}
